src/serialbox-c/Serializer.cpp

```diff
@@ -248,7 +260,8 @@ int serialboxSerializerAddField2(serialboxSerializer_t* serializer, const char*
     int rank =
         (iSize > 0 ? 1 : 0) + (jSize > 0 ? 1 : 0) + (kSize > 0 ? 1 : 0) + (lSize > 0 ? 1 : 0);
 
-    std::vector<int> dims{iSize, jSize, kSize, lSize};
+    auto dims = ::make_dims(iSize, jSize, kSize, lSize);
+
```

Comment:
Now you can compute the rank in a simpler way:

```
int rank = dims.size();
```

Reply:
Oh yes... Fixed.